docs/security/index.md

```diff
@@ -0,0 +1,453 @@
+# Security Plugins
+
+The `SecurityPlugin` interface provides an abstraction around the Telicent Core Platform's label based security model
+allowing applications to enforce the model without needing any direct knowledge of how the labels are interpreted and
+enforced.
+
+## Design
+
+In this section we first outline the design motivation behind the interfaces that are defined, see
+[Interfaces](#interfaces) for outlines of the various interfaces and their usages.  As with everything else in our Core
+Libraries this follows our general [Design Ethos](../design.md#design-ethos).
+
+### Label Format & Syntax
+
+As background, remember that all events flowing through Kafka within the Core Platform are expected to have a
+`Security-Label` header present that indicates the labels that apply to the data.  Additionally, RDF format events on
```

Comment:
What's the future for the `policyInformation` header? That currently contains either an EDH or IDH where the access control parts are converted into a security label. That was a stop-gap to allow us to develop federation and explore some of the principles you've laid out in this PR. In practice it works but is confusing for customers.

Reply:
I'd like to have a crack at implementing these interfaces with the scheme we used in the federation filter.

Reply:
So ideally the EDH/IDH is just put directly in the `Security-Label` header and you deploy a plugin that knows how to enforce the access criteria that the policy defines.

Yes our goal is to have probably 2 supported plugins - RDF-ABAC and EDH/IDH - which we build out as we start adopting this API and adapting it to address the inevitable implementation issues we discover

Reply:
Added some clarifications around this